In [1]:
# import libarary

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
df = pd.read_csv('zomato.csv',encoding= 'latin1')

In [3]:
df.head()

,Restaurant ID,Restaurant Name,Country Code,City,Address,Locality,Locality Verbose,Longitude,Latitude,Cuisines,...,Currency,Has Table booking,Has Online delivery,Is delivering now,Switch to order menu,Price range,Aggregate rating,Rating color,Rating text,Votes
0,6317637,Le Petit Souffle,162,Makati City,"Third Floor, Century City Mall, Kalayaan Avenu...","Century City Mall, Poblacion, Makati City","Century City Mall, Poblacion, Makati City, Mak...",121.027535,14.565443,"French, Japanese, Desserts",...,Botswana Pula(P),Yes,No,No,No,3,4.8,Dark Green,Excellent,314
1,6304287,Izakaya Kikufuji,162,Makati City,"Little Tokyo, 2277 Chino Roces Avenue, Legaspi...","Little Tokyo, Legaspi Village, Makati City","Little Tokyo, Legaspi Village, Makati City, Ma...",121.014101,14.553708,Japanese,...,Botswana Pula(P),Yes,No,No,No,3,4.5,Dark Green,Excellent,591
2,6300002,Heat - Edsa Shangri-La,162,Mandaluyong City,"Edsa Shangri-La, 1 Garden Way, Ortigas, Mandal...","Edsa Shangri-La, Ortigas, Mandaluyong City","Edsa Shangri-La, Ortigas, Mandaluyong City, Ma...",121.056831,14.581404,"Seafood, Asian, Filipino, Indian",...,Botswana Pula(P),Yes,No,No,No,4,4.4,Green,Very Good,270
3,6318506,Ooma,162,Mandaluyong City,"Third Floor, Mega Fashion Hall, SM Megamall, O...","SM Megamall, Ortigas, Mandaluyong City","SM Megamall, Ortigas, Mandaluyong City, Mandal...",121.056475,14.585318,"Japanese, Sushi",...,Botswana Pula(P),No,No,No,No,4,4.9,Dark Green,Excellent,365
4,6314302,Sambo Kojin,162,Mandaluyong City,"Third Floor, Mega Atrium, SM Megamall, Ortigas...","SM Megamall, Ortigas, Mandaluyong City","SM Megamall, Ortigas, Mandaluyong City, Mandal...",121.057508,14.584450,"Japanese, Korean",...,Botswana Pula(P),Yes,No,No,No,4,4.8,Dark Green,Excellent,229


In [4]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 9551 entries, 0 to 9550
Data columns (total 21 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Restaurant ID         9551 non-null   int64  
 1   Restaurant Name       9551 non-null   str    
 2   Country Code          9551 non-null   int64  
 3   City                  9551 non-null   str    
 4   Address               9551 non-null   str    
 5   Locality              9551 non-null   str    
 6   Locality Verbose      9551 non-null   str    
 7   Longitude             9551 non-null   float64
 8   Latitude              9551 non-null   float64
 9   Cuisines              9542 non-null   str    
 10  Average Cost for two  9551 non-null   int64  
 11  Currency              9551 non-null   str    
 12  Has Table booking     9551 non-null   str    
 13  Has Online delivery   9551 non-null   str    
 14  Is delivering now     9551 non-null   str    
 15  Switch to order menu  9551 non-n

In [5]:
df.duplicated().sum()

np.int64(0)

In [6]:
df.describe()

,Restaurant ID,Country Code,Longitude,Latitude,Average Cost for two,Price range,Aggregate rating,Votes
count,9.551000e+03,9551.000000,9551.000000,9551.000000,9551.000000,9551.000000,9551.000000,9551.000000
mean,9.051128e+06,18.365616,64.126574,25.854381,1199.210763,1.804837,2.666370,156.909748
std,8.791521e+06,56.750546,41.467058,11.007935,16121.183073,0.905609,1.516378,430.169145
min,5.300000e+01,1.000000,-157.948486,-41.330428,0.000000,1.000000,0.000000,0.000000
25%,3.019625e+05,1.000000,77.081343,28.478713,250.000000,1.000000,2.500000,5.000000
50%,6.004089e+06,1.000000,77.191964,28.570469,400.000000,2.000000,3.200000,31.000000
75%,1.835229e+07,1.000000,77.282006,28.642758,700.000000,2.000000,3.700000,131.000000
max,1.850065e+07,216.000000,174.832089,55.976980,800000.000000,4.000000,4.900000,10934.000000


In [7]:
df.isnull().sum()

Restaurant ID           0
Restaurant Name         0
Country Code            0
City                    0
Address                 0
Locality                0
Locality Verbose        0
Longitude               0
Latitude                0
Cuisines                9
Average Cost for two    0
Currency                0
Has Table booking       0
Has Online delivery     0
Is delivering now       0
Switch to order menu    0
Price range             0
Aggregate rating        0
Rating color            0
Rating text             0
Votes                   0
dtype: int64

Exploration Summary

* we have a dataframe consisting of 9551 rows and 21 columns

* our dataset looks a bit tidy with no NaNs nor duplicated values.

* Overview,'locality Verbose','Rating Color' and 'Rating text','Switch to order menu'and 'Latitude','Longitude' 
  wouldn't be so useful during analysis,so we'll drop them,

* processing applied to the 'Cuisines' column to remove extra whitespaces

* we'd split into a list and then explode our dataframe to have only one Cuisines per row for ezch restaurant.

* process the 'Address' column to remove newline characters and extra spaces.

* the dataset had 9551 entries,and missing values in the Cuisines column were removed to ensure clean data

In [8]:
df.head()

,Restaurant ID,Restaurant Name,Country Code,City,Address,Locality,Locality Verbose,Longitude,Latitude,Cuisines,...,Currency,Has Table booking,Has Online delivery,Is delivering now,Switch to order menu,Price range,Aggregate rating,Rating color,Rating text,Votes
0,6317637,Le Petit Souffle,162,Makati City,"Third Floor, Century City Mall, Kalayaan Avenu...","Century City Mall, Poblacion, Makati City","Century City Mall, Poblacion, Makati City, Mak...",121.027535,14.565443,"French, Japanese, Desserts",...,Botswana Pula(P),Yes,No,No,No,3,4.8,Dark Green,Excellent,314
1,6304287,Izakaya Kikufuji,162,Makati City,"Little Tokyo, 2277 Chino Roces Avenue, Legaspi...","Little Tokyo, Legaspi Village, Makati City","Little Tokyo, Legaspi Village, Makati City, Ma...",121.014101,14.553708,Japanese,...,Botswana Pula(P),Yes,No,No,No,3,4.5,Dark Green,Excellent,591
2,6300002,Heat - Edsa Shangri-La,162,Mandaluyong City,"Edsa Shangri-La, 1 Garden Way, Ortigas, Mandal...","Edsa Shangri-La, Ortigas, Mandaluyong City","Edsa Shangri-La, Ortigas, Mandaluyong City, Ma...",121.056831,14.581404,"Seafood, Asian, Filipino, Indian",...,Botswana Pula(P),Yes,No,No,No,4,4.4,Green,Very Good,270
3,6318506,Ooma,162,Mandaluyong City,"Third Floor, Mega Fashion Hall, SM Megamall, O...","SM Megamall, Ortigas, Mandaluyong City","SM Megamall, Ortigas, Mandaluyong City, Mandal...",121.056475,14.585318,"Japanese, Sushi",...,Botswana Pula(P),No,No,No,No,4,4.9,Dark Green,Excellent,365
4,6314302,Sambo Kojin,162,Mandaluyong City,"Third Floor, Mega Atrium, SM Megamall, Ortigas...","SM Megamall, Ortigas, Mandaluyong City","SM Megamall, Ortigas, Mandaluyong City, Mandal...",121.057508,14.584450,"Japanese, Korean",...,Botswana Pula(P),Yes,No,No,No,4,4.8,Dark Green,Excellent,229


Dropping The Columns

In [9]:
cols = ['Locality Verbose','Rating color','Rating text','Switch to order menu','Latitude','Longitude']

In [10]:
df.drop(cols,axis = 1, inplace = True)

df.columns

Index(['Restaurant ID', 'Restaurant Name', 'Country Code', 'City', 'Address',
       'Locality', 'Cuisines', 'Average Cost for two', 'Currency',
       'Has Table booking', 'Has Online delivery', 'Is delivering now',
       'Price range', 'Aggregate rating', 'Votes'],
      dtype='str')

In [11]:
df.head()

,Restaurant ID,Restaurant Name,Country Code,City,Address,Locality,Cuisines,Average Cost for two,Currency,Has Table booking,Has Online delivery,Is delivering now,Price range,Aggregate rating,Votes
0,6317637,Le Petit Souffle,162,Makati City,"Third Floor, Century City Mall, Kalayaan Avenu...","Century City Mall, Poblacion, Makati City","French, Japanese, Desserts",1100,Botswana Pula(P),Yes,No,No,3,4.8,314
1,6304287,Izakaya Kikufuji,162,Makati City,"Little Tokyo, 2277 Chino Roces Avenue, Legaspi...","Little Tokyo, Legaspi Village, Makati City",Japanese,1200,Botswana Pula(P),Yes,No,No,3,4.5,591
2,6300002,Heat - Edsa Shangri-La,162,Mandaluyong City,"Edsa Shangri-La, 1 Garden Way, Ortigas, Mandal...","Edsa Shangri-La, Ortigas, Mandaluyong City","Seafood, Asian, Filipino, Indian",4000,Botswana Pula(P),Yes,No,No,4,4.4,270
3,6318506,Ooma,162,Mandaluyong City,"Third Floor, Mega Fashion Hall, SM Megamall, O...","SM Megamall, Ortigas, Mandaluyong City","Japanese, Sushi",1500,Botswana Pula(P),No,No,No,4,4.9,365
4,6314302,Sambo Kojin,162,Mandaluyong City,"Third Floor, Mega Atrium, SM Megamall, Ortigas...","SM Megamall, Ortigas, Mandaluyong City","Japanese, Korean",1500,Botswana Pula(P),Yes,No,No,4,4.8,229


In [12]:
df['Cuisines'].head()

0          French, Japanese, Desserts
1                            Japanese
2    Seafood, Asian, Filipino, Indian
3                     Japanese, Sushi
4                    Japanese, Korean
Name: Cuisines, dtype: str

In [13]:
# remove exrta whitespaces 

df['Cuisines'] = df['Cuisines'].str.strip().str.replace(', ',',')

In [14]:
df['Cuisines'].head()

0         French,Japanese,Desserts
1                         Japanese
2    Seafood,Asian,Filipino,Indian
3                   Japanese,Sushi
4                  Japanese,Korean
Name: Cuisines, dtype: str

# we'd split into a list and then explode our dataframe to have only one Cuisines per row for ezch restaurant.

In [15]:
df['Cuisines'] = df['Cuisines'].str.split(', ')

df = df.explode('Cuisines').reset_index(drop = True)

In [21]:
df.head()

,Restaurant ID,Restaurant Name,Country Code,City,Address,Locality,Cuisines,Average Cost for two,Currency,Has Table booking,Has Online delivery,Is delivering now,Price range,Aggregate rating,Votes
0,6317637,Le Petit Souffle,162,Makati City,"Third Floor, Century City Mall, Kalayaan Avenu...","Century City Mall, Poblacion, Makati City","French,Japanese,Desserts",1100,Botswana Pula(P),Yes,No,No,3,4.8,314
1,6304287,Izakaya Kikufuji,162,Makati City,"Little Tokyo, 2277 Chino Roces Avenue, Legaspi...","Little Tokyo, Legaspi Village, Makati City",Japanese,1200,Botswana Pula(P),Yes,No,No,3,4.5,591
2,6300002,Heat - Edsa Shangri-La,162,Mandaluyong City,"Edsa Shangri-La, 1 Garden Way, Ortigas, Mandal...","Edsa Shangri-La, Ortigas, Mandaluyong City","Seafood,Asian,Filipino,Indian",4000,Botswana Pula(P),Yes,No,No,4,4.4,270
3,6318506,Ooma,162,Mandaluyong City,"Third Floor, Mega Fashion Hall, SM Megamall, O...","SM Megamall, Ortigas, Mandaluyong City","Japanese,Sushi",1500,Botswana Pula(P),No,No,No,4,4.9,365
4,6314302,Sambo Kojin,162,Mandaluyong City,"Third Floor, Mega Atrium, SM Megamall, Ortigas...","SM Megamall, Ortigas, Mandaluyong City","Japanese,Korean",1500,Botswana Pula(P),Yes,No,No,4,4.8,229


In [18]:
df['Address'].head()

0    Third Floor, Century City Mall, Kalayaan Avenu...
1    Little Tokyo, 2277 Chino Roces Avenue, Legaspi...
2    Edsa Shangri-La, 1 Garden Way, Ortigas, Mandal...
3    Third Floor, Mega Fashion Hall, SM Megamall, O...
4    Third Floor, Mega Atrium, SM Megamall, Ortigas...
Name: Address, dtype: str

In [20]:
# remove newline from Address column

df['Address'].str.replace(r'\\n','')

0       Third Floor, Century City Mall, Kalayaan Avenu...
1       Little Tokyo, 2277 Chino Roces Avenue, Legaspi...
2       Edsa Shangri-La, 1 Garden Way, Ortigas, Mandal...
3       Third Floor, Mega Fashion Hall, SM Megamall, O...
4       Third Floor, Mega Atrium, SM Megamall, Ortigas...
                              ...                        
9546    Kemankeô Karamustafa Paôa Mahallesi, RÛ±htÛ±...
9547    Koôuyolu Mahallesi, Muhittin íìstí_ndaÛô Cadd...
9548    Kuruí_eôme Mahallesi, Muallim Naci Caddesi, N...
9549    Kuruí_eôme Mahallesi, Muallim Naci Caddesi, N...
9550    CafeaÛôa Mahallesi, BademaltÛ± Sokak, No 21/B,...
Name: Address, Length: 9551, dtype: str

In [23]:
df.shape

(9551, 15)

In [25]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 9551 entries, 0 to 9550
Data columns (total 15 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Restaurant ID         9551 non-null   int64  
 1   Restaurant Name       9551 non-null   str    
 2   Country Code          9551 non-null   int64  
 3   City                  9551 non-null   str    
 4   Address               9551 non-null   str    
 5   Locality              9551 non-null   str    
 6   Cuisines              9542 non-null   str    
 7   Average Cost for two  9551 non-null   int64  
 8   Currency              9551 non-null   str    
 9   Has Table booking     9551 non-null   str    
 10  Has Online delivery   9551 non-null   str    
 11  Is delivering now     9551 non-null   str    
 12  Price range           9551 non-null   int64  
 13  Aggregate rating      9551 non-null   float64
 14  Votes                 9551 non-null   int64  
dtypes: float64(1), int64(5), str(9)


In [29]:
# drop missing values

df.dropna(subset = ['Cuisines'],inplace = True)

In [30]:
df.info()

<class 'pandas.DataFrame'>
Index: 9542 entries, 0 to 9550
Data columns (total 15 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Restaurant ID         9542 non-null   int64  
 1   Restaurant Name       9542 non-null   str    
 2   Country Code          9542 non-null   int64  
 3   City                  9542 non-null   str    
 4   Address               9542 non-null   str    
 5   Locality              9542 non-null   str    
 6   Cuisines              9542 non-null   str    
 7   Average Cost for two  9542 non-null   int64  
 8   Currency              9542 non-null   str    
 9   Has Table booking     9542 non-null   str    
 10  Has Online delivery   9542 non-null   str    
 11  Is delivering now     9542 non-null   str    
 12  Price range           9542 non-null   int64  
 13  Aggregate rating      9542 non-null   float64
 14  Votes                 9542 non-null   int64  
dtypes: float64(1), int64(5), str(9)
memor